### PR TITLE
set default CONTENT_TYPE when it is absent in multipart request body.

### PR DESCRIPTION
**Pull request by houdejun214**
Set default CONTENT_TYPE when it is absent in multipart request body

Motivation:

I am use netty as a http server, it fail to decode some POST request when the request absent Content-Type in the multipart/form-data body.

for example: I use this code (base on python `requests` library) to produce some http request,
```python
r = requests.post('http://testserverxxx/post', files={'file': open('Pink_Birkin_bag.jpg', 'rb')})
```

It will lead this decoder exception and can’t decode the request successfuly
```java
io.netty.handler.codec.http.multipart.HttpPostRequestDecoder$ErrorDataDecoderException: Content-Type is absent but required
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.getFileUpload(HttpPostMultipartRequestDecoder.java:838) ~[dependency-all-1.2.2-shaded.jar:na]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.decodeMultipart(HttpPostMultipartRequestDecoder.java:532) ~[dependency-all-1.2.2-shaded.jar:na]
    at io.netty.handler.codec.http.multipart.HttpPostMultipartRequestDecoder.findMultipartDisposition(HttpPostMultipartRequestDecoder.java:771) ~[dependency-all-1.2.2-shaded.jar:na]
```

Modifications:

Set content_type with default `application/octet-stream` to parse the uploaded file data when the Content-Type is absent in multipart request body

Result:

Can decode the http request as normal.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@houdejun214 can you please add a unit test, sign our icla [1] and adjust commit message to match our template [2] ?

[1] http://netty.io/s/icla
[2] http://netty.io/wiki/writing-a-commit-message.html

**Comment by houdejun214** [comment]
@normanmaurer , Ok,  I will follow this.

**Comment by houdejun214** [comment]
@normanmaurer updated, please check.

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java [hunk 1]
```diff
@@ -847,9 +844,13 @@ protected InterfaceHttpData getFileUpload(String delimiter) {
                 size = 0;
             }
             try {
+                String contentType = HttpPostBodyUtil.DEFAULT_BINARY_CONTENT_TYPE;
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostMultipartRequestDecoder.java]: could you move the assignment into an else block ?

**Comment by normanmaurer** [comment]
Cherry-picked into 4.0 (aa43c10403fb11dd25a764878d0306de12d1d341) and 4.1 (a6fd8a96bf54fb369927c0375f90502c2ac3d41e) after squashing the commits

**Pull request closed without merging**